### PR TITLE
[fix](mysql)Support COM_CHANGE_USER and other mysql command.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Support COM_CHANGE_USER and other mysql command.

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run cloud_p0

File: fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java [hunk 1]
```diff
@@ -249,6 +271,110 @@ private void handleFieldList() throws ConnectionException {
...
+        MysqlProto.readInt2(packetBuf);
+        String authPluginName = new String(MysqlProto.readNulTerminateString(packetBuf));
+
+        // Send Protocol::AuthSwitchRequest to client if auth plugin name is not mysql_native_password
+        if (!MysqlHandshakePacket.AUTH_PLUGIN_NAME.equals(authPluginName)) {
+            MysqlChannel channel = ctx.mysqlChannel;
+            MysqlSerializer serializer = MysqlSerializer.newInstance();
+            serializer.writeInt1((byte) 0xfe);
+            serializer.writeNulTerminateString(MysqlHandshakePacket.AUTH_PLUGIN_NAME);
+            serializer.writeBytes(authPluginData);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java:291]: authPluginData could be null, should we handle it before write?

File: fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java [hunk 2]
```diff
@@ -249,6 +271,110 @@ private void handleFieldList() throws ConnectionException {
...
+        ctx.setCurrentUserIdentity(currentUserIdentity.get(0));
+        ctx.setQualifiedUser(userName);
+
+        // Change default db if set.
+        if (Strings.isNullOrEmpty(db)) {
+            ctx.changeDefaultCatalog(InternalCatalog.INTERNAL_CATALOG_NAME);
+        } else {
+            String catalogName = null;
+            String dbName = null;
+            String[] dbNames = db.split("\\.");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java:325]: could `db` like
```
`123.456`.`789.012`
```
? 
do we need parse it with `NereidsParser.visitMultipartIdentifier`?

File: fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java (hunk 1 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java:293]: Fixed. Check not null before using it.

File: fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java (hunk 2 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java:327]: Tested, seems like `123.456` is not a valid database name. So I don't think we need to use visitMultipartIdentifier here.

**Comment by morrySnow** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run p0

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**